pkg/network/ovn_kubernetes_dpu_host_test.go

```diff
@@ -182,3 +185,172 @@ func createTestRenderData(ovnNodeMode string) render.RenderData {
+	g.Expect(ds.Spec.Template.Spec.Affinity.NodeAffinity).NotTo(BeNil(), "Should have node affinity")
+	g.Expect(ds.Spec.Template.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution).NotTo(BeNil(), "Should have required node affinity")
+
+	terms := ds.Spec.Template.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms
+	g.Expect(terms).NotTo(BeEmpty(), "Should have node selector terms")
+
+	matchExpressions := terms[0].MatchExpressions
+	g.Expect(matchExpressions).NotTo(BeEmpty(), "Should have match expressions")
+
+	for _, expr := range matchExpressions {
+		if expr.Key == label {
+			if expr.Operator == corev1.NodeSelectorOpIn {
+				g.Expect(len(expr.Values)).To(Equal(1), "In operator should have exactly one value")
+				return expr.Operator, expr.Values[0]
+			} else {
+				return expr.Operator, ""
+			}
+		}
+	}
+
+	return corev1.NodeSelectorOpDoesNotExist, ""
+}
+
+// TestOVNKubernetesNodeSelectorOperator tests that the node selector operator works correctly with label values of different Full/SmartNIC/DPU modes
+func TestOVNKubernetesNodeSelectorOperator(t *testing.T) {
+	templates := []struct {
+		name         string
+		templatePath string
+	}{
+		{
+			name:         "managed",
+			templatePath: "../../bindata/network/ovn-kubernetes/managed/ovnkube-node.yaml",
+		},
+		{
+			name:         "self-hosted",
+			templatePath: "../../bindata/network/ovn-kubernetes/self-hosted/ovnkube-node.yaml",
+		},
+	}
+
+	testCases := []struct {
+		name                 string
+		ovnNodeMode          string
+		dpuHostModeLabel     string
+		smartNicModeLabel    string
+		dpuModeLabel         string
+		expectedOperatorType corev1.NodeSelectorOperator
+		expectedValue        string // Expected value for In operator
+	}{
+		{
+			name:                 "dpu-host mode with key+value labels",
+			ovnNodeMode:          "dpu-host",
+			dpuHostModeLabel:     "network.operator.openshift.io/dpu-host=true",
+			smartNicModeLabel:    "network.operator.openshift.io/smart-nic=true",
+			dpuModeLabel:         "network.operator.openshift.io/dpu=true",
+			expectedOperatorType: corev1.NodeSelectorOpIn,
+			expectedValue:        "true",
+		},
+		{
+			name:                 "smart-nic mode with key+value labels",
+			ovnNodeMode:          "smart-nic",
+			dpuHostModeLabel:     "network.operator.openshift.io/dpu-host=true",
+			smartNicModeLabel:    "network.operator.openshift.io/smart-nic=true",
+			dpuModeLabel:         "network.operator.openshift.io/dpu=true",
+			expectedOperatorType: corev1.NodeSelectorOpIn,
+			expectedValue:        "true",
+		},
+		{
+			name:                 "dpu mode with key+value labels",
+			ovnNodeMode:          "dpu",
+			dpuHostModeLabel:     "network.operator.openshift.io/dpu-host=true",
+			smartNicModeLabel:    "network.operator.openshift.io/smart-nic=true",
+			dpuModeLabel:         "network.operator.openshift.io/dpu=true",
+			expectedOperatorType: corev1.NodeSelectorOpDoesNotExist,
```

Comment:
So here actually `expectedOperatorType` is never checked for `tc.ovnNodeMode == "dpu"`, but ok, we're setting our expectations explicitly.